### PR TITLE
[Amoro-2625] Intruduce new module structure

### LINKED ISSUE #2626: [Improvement]: Amoro module structure and dependencies improvement
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Refer to #2622 

### How should we improve?

The new structure shall be like:

image

But I won't delete metrics-reporter module because it is necessary for both server and optimizer metrics collecting

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by majin1102**
## Why are the changes needed?

Close #2626 .

## Brief change log

#2622 
- Reverse ams-api and core module dependency
  - Migrate thrift definitions and gen-java from ams-api to core
  - Migrate TableFormat and Catalog classes from ams-api to core
  - Modify dependencies in POMs
- Rename mixed module to mixed-format for better readibility

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (docs)

### CODE REVIEW

**Comment by majin1102** [comment]
Oh, no

CICD configuration needs to be modified for mixed drectory has been changed.

File: ams/api/src/test/java/com/netease/arctic/api/metrics/MetricDefineTest.java [hunk 1]
```diff
@@ -43,7 +44,7 @@ public void testEquals(MetricDefine target, boolean expectedEquality) {
       assertEquals(source.hashCode(), target.hashCode(), "MetricNames hash code should be equal");
     } else {
       assertNotEquals(source, target, "MetricNames should not be equal");
-      assertNotEquals(
+      Assertions.assertNotEquals(
```
Comment (zhoujinsong) [inline ams/api/src/test/java/com/netease/arctic/api/metrics/MetricDefineTest.java:47]: Is this change necessary?

File: ams/api/src/test/java/com/netease/arctic/api/metrics/TestMetricDefine.java [hunk 2]
```diff
@@ -44,7 +45,7 @@ void testEquals(MetricDefine target, boolean expectedEquality) {
       assertEquals(source.hashCode(), target.hashCode(), "MetricNames hash code should be equal");
     } else {
       assertNotEquals(source, target, "MetricNames should not be equal");
-      assertNotEquals(
+      Assertions.assertNotEquals(
```
Comment (zhoujinsong) [inline ams/api/src/test/java/com/netease/arctic/api/metrics/TestMetricDefine.java:48]: Is this change necessary?

File: core/src/main/java/com/netease/arctic/catalog/MixedTables.java [hunk 3]
```diff
@@ -18,23 +18,13 @@
...
-import com.netease.arctic.table.ChangeTable;
-import com.netease.arctic.table.KeyedTable;
-import com.netease.arctic.table.PrimaryKeySpec;
-import com.netease.arctic.table.TableIdentifier;
-import com.netease.arctic.table.TableMetaStore;
-import com.netease.arctic.table.TableProperties;
-import com.netease.arctic.table.UnkeyedTable;
-import com.netease.arctic.utils.CatalogUtil;
+import com.netease.arctic.properties.MetaTableProperties;
+import com.netease.arctic.table.*;
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/catalog/MixedTables.java:26]: We should try to avoid importing the entire package whenever possible.

**Review by zhoujinsong — COMMENTED** [review]
Thanks a lot for your contribution!
I left some comments, besides the following areas can be improved:

- You may want to change some configuration in `.github/labeler.yml` due to the module change.
- You may want to change some contents about `mixed` in `docs/admin-guides/deployment.md` due to the module change.
- You may want to change some contents about `mixed` in `CONTRIBUTING.md` due to the module change.

File: ams/api/src/test/java/com/netease/arctic/api/metrics/MetricDefineTest.java (hunk 1 above)
Comment (majin1102) [reply ams/api/src/test/java/com/netease/arctic/api/metrics/MetricDefineTest.java:47]: This should be auto-generated when optimizing imports
I will fix it.

File: ams/api/src/test/java/com/netease/arctic/api/metrics/TestMetricDefine.java (hunk 2 above)
Comment (majin1102) [reply ams/api/src/test/java/com/netease/arctic/api/metrics/TestMetricDefine.java:48]: done

File: core/src/main/java/com/netease/arctic/catalog/MixedTables.java (hunk 3 above)
Comment (majin1102) [reply core/src/main/java/com/netease/arctic/catalog/MixedTables.java:26]: done

**Comment by majin1102** [comment]
> * CONTRIBUTING.md

Thanks for reviewing so quick.

The work has been done.
